File: front/src/pages/Jydboot/SysMenu/SysMenu.js
import React, { PureComponent } from 'react';
import { connect } from 'dva';
import { Button, Col, Form, Input, Row } from 'antd';
import Panel from '../../../components/Panel';
import { SYSMENU_LIST } from '../../../actions/sysMenu';
import Grid from '../../../components/Sword/Grid';

const FormItem = Form.Item;

@connect(({ sysMenu, loading }) => ({
  sysMenu,
  loading: loading.models.sysMenu,
}))
@Form.create()
class SysMenu extends PureComponent {
  // ============ 查询 ===============
  handleSearch = params => {
    const { dispatch } = this.props;
    dispatch(SYSMENU_LIST(params));
  };

  // ============ 查询表单 ===============
  renderSearchForm = onReset => {
    const { form } = this.props;
    const { getFieldDecorator } = form;

    return (
      <Row gutter={{ md: 8, lg: 24, xl: 48 }}>
        <Col md={6} sm={24}>
          <FormItem label="查询名称">
            {getFieldDecorator('name')(<Input placeholder="查询名称" />)}
          </FormItem>
        </Col>
        <Col>
          <div style={{ float: 'right' }}>
            <Button type="primary" htmlType="submit">
              查询
            </Button>
            <Button style={{ marginLeft: 8 }} onClick={onReset}>
              重置
            </Button>
          </div>
        </Col>
      </Row>
    );
  };

  render() {
    const code = 'sysMenu';

    const {
      form,
      loading,
      sysMenu: { data },
    } = this.props;

    const columns = [
      {
        title: 'ID',
        dataIndex: 'id',
      },
      {
        title: '创建时间',
        dataIndex: 'createTime',
      },
      {
        title: '更新时间',
        dataIndex: 'updateTime',
      },
      {
        title: '创建人',
        dataIndex: 'createUser',
      },
      {
        title: '更新人',
        dataIndex: 'updateUser',
      },
      {
        title: '状态 0正常 1锁定',
        dataIndex: 'status',
      },
      {
        title: '删除标记 0正常 1删除',
        dataIndex: 'deleted',
      },
      {
        title: '权限编码',
        dataIndex: 'code',
      },
      {
        title: '权限名称',
        dataIndex: 'name',
      },
      {
        title: '路由',
        dataIndex: 'path',
      },
      {
        title: '图标',
        dataIndex: 'source',
      },
      {
        title: '备注',
        dataIndex: 'remark',
      },
      {
        title: '序号',
        dataIndex: 'sort',
      },
      {
        title: '1菜单2按钮3数据',
        dataIndex: 'category',
      },
      {
        title: '父级ID',
        dataIndex: 'parentId',
      },
      {
        title: '别名',
        dataIndex: 'alias',
      },
      {
        title: '操作按钮类型',
        dataIndex: 'action',
      },
      {
        title: '是否隐藏',
        dataIndex: 'isHidden',
      },
      {
        title: '是否打开新页面',
        dataIndex: 'isOpen',
      },
    ];

    return (
      <Panel>
        <Grid
          code={code}
          form={form}
          onSearch={this.handleSearch}
          renderSearchForm={this.renderSearchForm}
          loading={loading}
          data={data}
          columns={columns}
        />
      </Panel>
    );
  }
}
export default SysMenu;

File: src/main/java/com/jyd/entity/SysMenu.java
package com.jyd.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * 权限实体类
 *
 * @author
 * @since 2023-12-01
 */
@Data
@TableName("t_sys_menu")
@ApiModel(value = "SysMenu对象", description = "权限")
public class SysMenu implements Serializable {

	private static final long serialVersionUID = 1L;

	/**
	* ID
	*/
		@ApiModelProperty(value = "ID")
		private String id;
	/**
	* 创建时间
	*/
		@ApiModelProperty(value = "创建时间")
		private Date createTime;
	/**
	* 更新时间
	*/
		@ApiModelProperty(value = "更新时间")
		private Date updateTime;
	/**
	* 创建人
	*/
		@ApiModelProperty(value = "创建人")
		private String createUser;
	/**
	* 更新人
	*/
		@ApiModelProperty(value = "更新人")
		private String updateUser;
	/**
	* 状态 0正常 1锁定
	*/
		@ApiModelProperty(value = "状态 0正常 1锁定")
		private Integer status;
	/**
	* 删除标记 0正常 1删除
	*/
		@ApiModelProperty(value = "删除标记 0正常 1删除")
		private Integer deleted;
	/**
	* 权限编码
	*/
		@ApiModelProperty(value = "权限编码")
		private String code;
	/**
	* 权限名称
	*/
		@ApiModelProperty(value = "权限名称")
		private String name;
	/**
	* 路由
	*/
		@ApiModelProperty(value = "路由")
		private String path;
	/**
	* 图标
	*/
		@ApiModelProperty(value = "图标")
		private String source;
	/**
	* 备注
	*/
		@ApiModelProperty(value = "备注")
		private String remark;
	/**
	* 序号
	*/
		@ApiModelProperty(value = "序号")
		private Integer sort;
	/**
	* 1菜单2按钮3数据
	*/
		@ApiModelProperty(value = "1菜单2按钮3数据")
		private Integer category;
	/**
	* 父级ID
	*/
		@ApiModelProperty(value = "父级ID")
		private String parentId;
	/**
	* 别名
	*/
		@ApiModelProperty(value = "别名")
		private String alias;
	/**
	* 操作按钮类型
	*/
		@ApiModelProperty(value = "操作按钮类型")
		private Integer action;

	/**
	* 是否打开新页面
	*/
		@ApiModelProperty(value = "是否打开新页面")
		private Integer isOpen;


}
